github/pulls.go

```diff
@@ -134,6 +135,29 @@ func (s *PullRequestsService) Get(owner string, repo string, number int) (*PullR
 	return pull, resp, err
 }
 
+// GetRaw gets raw (diff or patch) format of a pull request.
+func (s *PullRequestsService) GetRaw(owner string, repo string, number int, opt GetRawOptions) ([]byte, *Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/pulls/%d", owner, repo, number)
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return []byte{}, nil, err
+	}
+
+	if opt.Type == Diff {
+		req.Header.Set("Accept", mediaTypeV3Diff)
+	} else if opt.Type == Patch {
+		req.Header.Set("Accept", mediaTypeV3Patch)
+	}
```

Comment:
Two things here.

1. A `switch` statement would be a good fit here.
2. How about handling when `opt.Type` value is not any of the supported types? I think returning an error would be most appropriate. Currently, it would just perform unspecified actions.